var/spack/repos/builtin/packages/gcc-runtime/package.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import re
+
+from spack.package import *
+from spack.util.elf import parse_elf
+
+
+class GccRuntime(Package):
+    """Package for GCC compiler runtime libraries"""
+
+    homepage = "https://gcc.gnu.org"
+    has_code = False
+
+    maintainers("haampie")
+
+    license("GPL-3.0-only")
+
+    requires("%gcc")
+
+    for v in [
+        "13.2",
+        "13.1",
+        "12.3",
+        "12.2",
+        "12.1",
+        "11.4",
+        "11.3",
+        "11.2",
+        "11.1",
+        "10.5",
+        "10.4",
+        "10.3",
+        "10.2",
+        "10.1",
+        "9.5",
+        "9.4",
+        "9.3",
+        "9.2",
+        "9.1",
+        "8.5",
+        "8.4",
+        "8.3",
+        "8.2",
+        "8.1",
+        "7.5",
+        "7.4",
+        "7.3",
+        "7.2",
+        "7.1",
+        "6.5",
+        "6.4",
+        "6.3",
+        "6.2",
+        "6.1",
+        "5.5",
+        "5.4",
+        "5.3",
+        "5.2",
+        "5.1",
+        "4.9.4",
+        "4.9.3",
+        "4.9.2",
+        "4.9.1",
+        "4.9.0",
+        "4.8.5",
+        "4.8.4",
+        "4.8.3",
+        "4.8.2",
+        "4.8.1",
+        "4.8.0",
+        "4.7.4",
+        "4.7.3",
+        "4.7.2",
+        "4.7.1",
+        "4.7.0",
+        "4.6.4",
+        "4.6.3",
+        "4.6.2",
+        "4.6.1",
+        "4.6.0",
+        "4.5.4",
+        "4.5.3",
+        "4.5.2",
+        "4.5.1",
+        "4.5.0",
+    ]:
+        version(v)
+        requires(f"%gcc@{v}", when=f"@{v}")
```

Comment:
Wondering if this is the right place for the `requires`. This requirement is on the `gcc-runtime` node itself, and is not related to the compiler choice in other nodes. That means we can concretize:
```
hdf5~mpi%gcc@13.2.0 ^gcc-runtime%gcc@9.4.0
```
and get 
![hdf5](https://github.com/spack/spack/assets/4199709/56d9c0f1-df2a-41c7-a006-b109bd70d4f0)

This is a simple synthetic example, but wondering if the same might happen due to indirect rules imposing a requirement. Do we have ideas how we want to treat compatibilities between runtimes and compilers?

Reply:
This is no longer possible

Reply:
But `hdf5~mpi%gcc@10 ^gcc-runtime%gcc@13` is possible